_modules/vault.py

```diff
@@ -176,19 +177,22 @@ def check_cached_lease(path, cache_prefix='', **kwargs):
     vault_data = vault_client.read(cache_path)
 
     if vault_data:
-        log.debug('Loaded cached data for path %s', path)
         vault_data = vault_data['data']['value']
         lease = vault_client.get_lease(vault_data['lease_id'])
 
         if (lease and timedelta(seconds=lease['data']['ttl']) >
                 timedelta(**renewal_threshold)):
             lease_valid = True
-        else:
+        # Lease might sitll be in cache even though it has expired.
+        # This verifies that the lease is still valid and can be renewed.
+        elif lease and vault_data['lease_duration'] > 0:
             if vault_data['renewable']:
```

Comment:
Should we change this to pull the renewable attribute from the `lease` object that we get from Vault, rather than the cached data?